analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/RestrictDeserializedTypesBase.cs

```diff
@@ -78,10 +78,8 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         else if (UnsafeDeserialization(state, operation) is { } invocation)
         {
             var methodDeclaration = UnsafeMethodDeclaration(state, invocation.Instance);
-            var additionalLocations = methodDeclaration is not null
-                ? new[] { GetIdentifier(methodDeclaration).GetLocation() }
-                : Array.Empty<Location>();
-            ReportIssue(operation, additionalLocations, additionalLocations.ToProperties(SecondaryMessage), RestrictTypesMessage);
+            var additionalLocations = methodDeclaration is null ? [] : new[] { GetIdentifier(methodDeclaration).GetLocation().ToSecondary(SecondaryMessage) };
```

Comment:
```suggestion
            var additionalLocations = methodDeclaration is null ? [] : [ GetIdentifier(methodDeclaration).GetLocation().ToSecondary(SecondaryMessage) ];
```

Reply:
I tried this one too, but it cannot infer the underlying type (array vs list vs enumerable vs... ). The VB version would work here, as that one always creates array.